CMakeLists.txt

```diff
@@ -15,44 +15,91 @@ option(WITH_OTLP "Whether to include the OpenTelemetry Protocol in the SDK" OFF)
 option(WITH_PROMETHEUS "Whether to include the Prometheus Client in the SDK"
        OFF)
 
-option(WITH_TESTS "Whether to enable tests" ON)
+option(BUILD_TESTING "Whether to enable tests" ON)
 option(WITH_EXAMPLES "Whether to build examples" ON)
 
 set(WITH_PROTOBUF OFF)
-if(WITH_OTLP)
-  set(WITH_PROTOBUF ON)
-endif()
-
-if(WITH_TESTS)
-  include(CTest)
-endif()
 
 find_package(Threads)
 
+function(install_windows_deps)
+  # Bootstrap vcpkg from CMake and auto-install deps in case if we are missing
+  # deps on Windows
+  message("Installing build tools and dependencies...")
+  execute_process(
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/tools/setup-buildtools.cmd)
+  set(CMAKE_TOOLCHAIN_FILE
+      ${CMAKE_CURRENT_SOURCE_DIR}/tools/vcpkg/scripts/buildsystems/vcpkg.cmake)
+endfunction()
+
 if(MSVC)
   # Options for Visual C++ compiler: /Zc:__cplusplus - report an updated value
   # for recent C++ language standards. Without this option MSVC returns the
   # value of __cplusplus="199711L"
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:__cplusplus")
 endif()
 
+if(WITH_OTLP)
+  set(WITH_PROTOBUF ON)
+endif()
+
 if(WITH_PROTOBUF)
   set(protobuf_MODULE_COMPATIBLE ON)
-  find_package(Protobuf CONFIG NAMES protobuf)
-  # Older versions of protobuf don't use cmake config files.
+  find_package(Protobuf REQUIRED)
   if(NOT protobuf_FOUND)
+    if(WIN32)
+      install_windows_deps()
+    endif()
     find_package(Protobuf REQUIRED)
+    if(WIN32)
+      # Always use x64 protoc.exe
+      if(NOT EXISTS "${Protobuf_PROTOC_EXECUTABLE}")
+        set(Protobuf_PROTOC_EXECUTABLE
+            ${CMAKE_CURRENT_SOURCE_DIR}/tools/vcpkg/packages/protobuf_x64-windows/tools/protobuf/protoc.exe
+        )
+      endif()
+    endif()
+    # Latest Protobuf uses mixed case instead of uppercase
+    set(PROTOBUF_PROTOC_EXECUTABLE ${Protobuf_PROTOC_EXECUTABLE})
   endif()
+  message("PROTOBUF_PROTOC_EXECUTABLE=${PROTOBUF_PROTOC_EXECUTABLE}")
 endif()
 
 if(WITH_OTLP)
-  include(third_party/opentelemetry-proto/Protobuf.cmake)
+  include(cmake/opentelemetry-proto.cmake)
 endif()
 
+list(APPEND CMAKE_PREFIX_PATH "${CMAKE_BINARY_DIR}")
+
 if(BUILD_TESTING)
-  find_package(GTest REQUIRED)
-  find_package(benchmark REQUIRED)
+  include(CTest)
+  if(EXISTS ${CMAKE_BINARY_DIR}/lib/libgtest.a)
```

Comment:
This is needed to ensure that our GTest is compiled with matching compiler / STL. Otherwise if we try to use the sytem-lib GTest, we may experience a break if compiling with `gcc-4.8`, but the system GTest is compiled with `gcc-10` or alike.